tests/security/apparmor/aa_easyprof.pm

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: AppArmor profile generation made easy testing
+# Maintainer: Wes <whdu@suse.com>
+# Tags: poo#36895, tc#1621144
+
+use strict;
+use base "apparmortest";
+use testapi;
+use utils;
+use version_utils qw(is_sle is_leap);
+
+sub run {
+    my ($self) = @_;
+
+    my $aa_easyprof_jsonopt;
+    my $aa_easyprof_out;
+    my $aa_easyprof_tmp_json = "/tmp/manifest.json";
+    my $aa_easyprof_result   = "/tmp/output";
+    my @aa_easyprof_cmd;
+    my $aa_easyprof_cmd_convert;
+
+    # For new apparmor version, easyprof support convert from a manifest
+    # json file
+    if (is_sle('<15') or is_leap('<15.0')) {    # apparmor < 2.11.95
+        $aa_easyprof_out = $aa_easyprof_result;
+    }
+    else {                                      # apparmor >= 2.11.95
+        $aa_easyprof_out         = $aa_easyprof_tmp_json;
+        $aa_easyprof_jsonopt     = "--output-format=json";
+        $aa_easyprof_cmd_convert = "aa-easyprof --manifest=$aa_easyprof_out > $aa_easyprof_result";
+    }
+
+    @aa_easyprof_cmd = (
```

Comment:
what the point to create an array which only purpose is to be concatenated to a string ? Just create a string or if you worry about formating create a bash script file and put it in data folder , upload it into a SUT and execute

Reply:
The command string is different between two situations, which controlled by  _$aa_easyprof_jsonopt_. I do not want to write the similar code twice to make duplication.

Reply:
`$aa_easyprof_cmd="aa-easyprof  $aa_easyprof_jsonopt --template=user-application --policy-groups=opt-application,user-application --abstractions="python,audio" --read-path="/usr/share/foo/*"   --write-ath="/opt/foo/tmp/"         --write-path="/opt/foo/log/" --template-var="@{APPNAME}=foo"  --author="SUSE Tester"  --copyright="Copyright 2018, SUSE Tester" --comment="AppArmor is easy with aa-easyprof" /usr/bin/foo > $aa_easyprof_out"` 

no code ducplication , no array and `$aa_easyprof_jsonopt` is still in .

to avoid problems in string concatenation just update place where you declare variable to 
`my @aa_easyprof_cmd = " ";`

Reply:
ok maybe your version is more readable :)

Reply:
I found a way to make it simple and keep readability ...

Reply:
sure , let's leave it as it is . sorry for confusing you :\ . At the beginning it was looking like a good idea :)